### ISSUE TITLE
Dashboard with multiple graphs taking full CPU

### ISSUE THREAD

**Issue by paulbakker** [Graylog2/graylog2-server#2727]
I have a dashboard with 10 graphs on it. Loading this dashboard in Chrome or Firefox causes 100% cpu load, sometimes crashing the tab. The load doesn't lower anymore either.
With ~5 graphs everything seems fine, with more than that the issue starts to appear.

* Graylog Version: v2.0.3
* Browser version: Chrome Version 52.0.2743.116 (64-bit) / FireFox 

The following is the json for the dashboard to give an idea how it is configured.

<img width="1750" alt="screen shot 2016-08-22 at 20 02 52" src="https://cloud.githubusercontent.com/assets/109484/17865588/792ba2d8-68a3-11e6-84c4-184711afc8c1.png">

```
{
  "total": 1,
  "dashboards": [
    {
      "creator_user_id": "admin",
      "description": "paul-1",
      "created_at": "2016-08-22T16:33:37.669Z",
      "positions": {},
      "id": "57bb29618716a400086c42fc",
      "title": "paul-1",
      "widgets": [
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-hv449 gc time",
          "id": "58d225b5-12be-49b6-914c-e3e138e1f224",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "line",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: gc_time AND pod: bookmarks-5-hv449",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-hv449 garbage collects",
          "id": "c793ee25-977f-4fe3-8202-f8f1f31f44a2",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "bar",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: gc_collections AND pod: bookmarks-5-hv449",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-hv449 threads",
          "id": "d07d052f-eda1-4493-a834-c46ec5ecdcff",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "bar",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType:threads AND pod: bookmarks-5-hv449",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-hv449 memory",
          "id": "b6e0dc3d-01ba-4294-9409-cd39990e3132",
          "type": "STACKED_CHART",
          "config": {
            "renderer": "area",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "stream_id": "577670fae4b06ae4ca79846c",
            "series": [
              {
                "query": "metricType: memory_used AND pod: bookmarks-5-hv449",
                "field": "value",
                "statistical_function": "mean"
              },
              {
                "query": "metricType: memory_max AND pod: bookmarks-5-hv449",
                "field": "value",
                "statistical_function": "mean"
              },
              {
                "query": "metricType: memory_committed AND pod: bookmarks-5-hv449",
                "field": "value",
                "statistical_function": "mean"
              }
            ],
            "query": "",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-hv449 cpu load",
          "id": "34e7105a-0c3e-4e9d-a5f9-b93c764f3bbf",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "line",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: cpu_processload AND pod: bookmarks-5-hv449",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-kg62z memory",
          "id": "c499c854-55b2-4876-a48a-558a186b8aca",
          "type": "STACKED_CHART",
          "config": {
            "renderer": "area",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "stream_id": "577670fae4b06ae4ca79846c",
            "series": [
              {
                "query": "metricType: memory_used AND pod: bookmarks-5-kg62z",
                "field": "value",
                "statistical_function": "mean"
              },
              {
                "query": "metricType: memory_max AND pod: bookmarks-5-kg62z",
                "field": "value",
                "statistical_function": "mean"
              },
              {
                "query": "metricType: memory_committed AND pod: bookmarks-5-kg62z",
                "field": "value",
                "statistical_function": "mean"
              }
            ],
            "query": "",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-kg62z threads",
          "id": "8a583ac2-bad9-45bb-a3a8-f3ad77720fdc",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "bar",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType:threads AND pod: bookmarks-5-kg62z",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-kg62z cpu load",
          "id": "5d2046bd-99e8-4fb7-9684-f4591dadfe65",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "line",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: cpu_processload AND pod: bookmarks-5-kg62z",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-kg62z gc time",
          "id": "5a7427be-c2ef-40c1-a69b-7cfa554d7f5f",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "line",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: gc_time AND pod: bookmarks-5-kg62z",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-kg62z garbage collects",
          "id": "a6115bc8-4d62-4597-8a8f-01409f9b12ed",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "bar",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: gc_collections AND pod: bookmarks-5-kg62z",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-l9vru memory",
          "id": "e5b96a1f-3114-4b82-b61a-8819d937fafd",
          "type": "STACKED_CHART",
          "config": {
            "renderer": "area",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "stream_id": "577670fae4b06ae4ca79846c",
            "series": [
              {
                "query": "metricType: memory_used AND pod: bookmarks-5-l9vru",
                "field": "value",
                "statistical_function": "mean"
              },
              {
                "query": "metricType: memory_max AND pod: bookmarks-5-l9vru",
                "field": "value",
                "statistical_function": "mean"
              },
              {
                "query": "metricType: memory_committed AND pod: bookmarks-5-l9vru",
                "field": "value",
                "statistical_function": "mean"
              }
            ],
            "query": "",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-l9vru threads",
          "id": "cd1450d1-0d0e-4178-a60b-31cc0ef82bd5",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "bar",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType:threads AND pod: bookmarks-5-l9vru",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-l9vru garbage collects",
          "id": "73d42a4a-948d-490a-b77d-a3d06f10590a",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "bar",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: gc_collections AND pod: bookmarks-5-l9vru",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-l9vru cpu load",
          "id": "abca6381-f762-4c7e-9cbf-a080e31084ce",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "line",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: cpu_processload AND pod: bookmarks-5-l9vru",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-l9vru gc time",
          "id": "d12d325c-4c84-4afa-a94a-7611ad806eec",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "line",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: gc_time AND pod: bookmarks-5-l9vru",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-t6r96 memory",
          "id": "4cefa5c6-26c7-4264-a94f-ae492ffbf30c",
          "type": "STACKED_CHART",
          "config": {
            "renderer": "area",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "stream_id": "577670fae4b06ae4ca79846c",
            "series": [
              {
                "query": "metricType: memory_used AND pod: bookmarks-5-t6r96",
                "field": "value",
                "statistical_function": "mean"
              },
              {
                "query": "metricType: memory_max AND pod: bookmarks-5-t6r96",
                "field": "value",
                "statistical_function": "mean"
              },
              {
                "query": "metricType: memory_committed AND pod: bookmarks-5-t6r96",
                "field": "value",
                "statistical_function": "mean"
              }
            ],
            "query": "",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-t6r96 threads",
          "id": "c06da78e-4e27-4e25-bab2-7f4bb6eaa220",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "bar",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType:threads AND pod: bookmarks-5-t6r96",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-t6r96 garbage collects",
          "id": "96a1f9ac-f850-418f-ba04-3ce9e2d7a8dc",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "bar",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: gc_collections AND pod: bookmarks-5-t6r96",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-t6r96 gc time",
          "id": "98e1614c-94e9-4917-81db-202d60cadf67",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "line",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: gc_time AND pod: bookmarks-5-t6r96",
            "interval": "minute",
            "relative": 86400
          }
        },
        {
          "creator_user_id": "admin",
          "cache_time": 10,
          "description": "pod bookmarks-5-t6r96 cpu load",
          "id": "6d42e35e-f30e-4311-96d0-ebcc0204bcaf",
          "type": "FIELD_CHART",
          "config": {
            "valuetype": "max",
            "renderer": "line",
            "interpolation": "linear",
            "timerange": {
              "type": "relative",
              "range": 86400
            },
            "rangeType": "relative",
            "field": "value",
            "stream_id": "577670fae4b06ae4ca79846c",
            "query": "metricType: cpu_processload AND pod: bookmarks-5-t6r96",
            "interval": "minute",
            "relative": 86400
          }
        }
      ]
    }
  ]
}
```

**Comment by edmundoa** [comment]
Hi,

This issue has been fixed in #2281, and will be released in 2.1.0. Feel free to look at the [latest beta](https://www.graylog.org/blog/66-announcing-graylog-2-1-0-beta-4) and submit any feedback on it.

Thank you!

**Comment by paulbakker** [comment]
Excellent, thanks!
